linera-chain/src/manager.rs

```diff
@@ -288,11 +339,15 @@ impl MultiOwnerManager {
             }
         }
         if let Some(key_pair) = key_pair {
-            // Vote to validate.
             let BlockAndRound { block, round } = proposal.content;
             let executed_block = outcome.with(block);
-            let vote = Vote::new(HashedValue::new_validated(executed_block), round, key_pair);
-            self.pending = Some(vote);
+            // If this is a fast block, vote to confirm. Otherwise vote to validate.
+            let value = if round.is_fast() {
+                HashedValue::new_confirmed(executed_block)
+            } else {
+                HashedValue::new_validated(executed_block)
+            };
+            self.pending = Some(Vote::new(value, round, key_pair));
         }
     }
 
```

Comment:
Just checking: `into_confirmed` below seems to allow this function to vote for a block that was already confirmed.

Reply:
I didn't check for that because it wouldn't be unsafe to do it. But the worker calls `create_final_vote` only in `process_validated_block`, so that can't happen: In fast rounds, there are no validated blocks.